src/indexing.jl

```diff
@@ -377,3 +377,29 @@ Base.unsafe_view(A::AbstractArray, i1::StaticIndexing, indices::StaticIndexing..
 # the tuple indices has to have at least one element to prevent infinite
 # recursion when viewing a zero-dimensional array (see issue #705)
 Base.SubArray(A::AbstractArray, indices::Tuple{StaticIndexing, Vararg{StaticIndexing}}) = Base.SubArray(A, map(unwrap, indices))
+
+###########################################################
+# SDiagonal
+###########################################################
+
+# SDiagonal uses Cartesian indexing, and the canonical indexing methods shadow getindex for Diagonal
+# these are needed for ambiguity resolution
+@inline function getindex(D::SDiagonal, i::Int, j::Int)
```

Comment:
If this is just for ambiguity resolution, could you maybe just `invoke` the `LinearAlgebra` method?

Reply:
I've updated this now, thanks for the suggestion!